go/test/endtoend/onlineddl/scheduler/onlineddl_scheduler_test.go

```diff
@@ -671,82 +764,318 @@ func TestSchemaChange(t *testing.T) {
 	}
 }
 
+func TestSingleton(t *testing.T) {
+	defer cluster.PanicHandler(t)
+
+	require.Equal(t, 1, len(shards))
+
+	createParams := func(ddlStatement string, ddlStrategy string, executeStrategy string, migrationContext string, expectHint string, expectError string, skipWait bool) *testOnlineDDLStatementParams {
+		return &testOnlineDDLStatementParams{
+			ddlStatement:     ddlStatement,
+			ddlStrategy:      ddlStrategy,
+			executeStrategy:  executeStrategy,
+			migrationContext: migrationContext,
+			expectHint:       expectHint,
+			expectError:      expectError,
+			skipWait:         skipWait,
+		}
+	}
+
+	createRevertParams := func(revertUUID string, ddlStrategy string, executeStrategy string, migrationContext string, expectError string, skipWait bool) *testRevertMigrationParams {
+		return &testRevertMigrationParams{
+			revertUUID:       revertUUID,
+			executeStrategy:  executeStrategy,
+			ddlStrategy:      ddlStrategy,
+			migrationContext: migrationContext,
+			expectError:      expectError,
+			skipWait:         skipWait,
+		}
+	}
+
+	var (
+		tableName                         = `stress_test`
+		onlineSingletonDDLStrategy        = "online --singleton"
+		onlineSingletonContextDDLStrategy = "online --singleton-context"
+		createStatement                   = `
+		CREATE TABLE stress_test (
+			id bigint(20) not null,
+			rand_val varchar(32) null default '',
+			hint_col varchar(64) not null default 'just-created',
+			created_timestamp timestamp not null default current_timestamp,
+			updates int unsigned not null default 0,
+			PRIMARY KEY (id),
+			key created_idx(created_timestamp),
+			key updates_idx(updates)
+		) ENGINE=InnoDB
+	`
+		// We will run this query with "gh-ost --max-load=Threads_running=1"
+		alterTableThrottlingStatement = `
+		ALTER TABLE stress_test DROP COLUMN created_timestamp
+	`
+		multiAlterTableThrottlingStatement = `
+		ALTER TABLE stress_test ENGINE=InnoDB;
+		ALTER TABLE stress_test ENGINE=InnoDB;
+		ALTER TABLE stress_test ENGINE=InnoDB;
+	`
+		// A trivial statement which must succeed and does not change the schema
+		alterTableTrivialStatement = `
+		ALTER TABLE stress_test ENGINE=InnoDB
+	`
+		dropStatement = `
+		DROP TABLE stress_test
+	`
+		dropNonexistentTableStatement = `
+		DROP TABLE IF EXISTS t_non_existent
+	`
+		multiDropStatements = `DROP TABLE IF EXISTS t1; DROP TABLE IF EXISTS t2; DROP TABLE IF EXISTS t3;`
+	)
+
+	// func testOnlineDDLStatement(t *testing.T, ddlStatement string, ddlStrategy string, executeStrategy string,                            expectHint string, expectError string, skipWait bool) (uuid string) {
+	// func testOnlineDDLStatement(t *testing.T, alterStatement string, ddlStrategy string, executeStrategy string, migrationContext string, expectHint string, expectError string, skipWait bool) (uuid string) {
```

Comment:
Nit, but we should remove these I think.

Reply:
whoops. removed.